src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/RandomPartitioningConditionMapper.java

```diff
@@ -30,143 +38,272 @@
         this.fieldMapper = fieldMapper;
     }
 
-    /**
-     * Extracts literals referenced in primary and filter expressions and turns them into references to
-     * expression names and values.
-     */
-    @VisibleForTesting
-    void convertFieldNameLiteralsToExpressionNames(RequestWrapper request, Collection<FieldMapping> fieldMappings) {
-        request.setPrimaryExpression(
-            convertFieldNameLiteralsToExpressionNamesInternal(fieldMappings, request.getPrimaryExpression(), request));
-        request.setFilterExpression(
-            convertFieldNameLiteralsToExpressionNamesInternal(fieldMappings, request.getFilterExpression(), request));
+    @Override
+    public void applyForUpdate(RequestWrapper request) {
+        String updateExpression = request.getPrimaryExpression();
+        String conditionExpression = request.getFilterExpression();
+        Set<String> virtualFieldNames = tableMapping.getAllMappingsPerField().keySet();
+
+        updateExpression = convertFieldNameLiteralsToExpressionNames(updateExpression, request, virtualFieldNames);
+        validateNotUpdatingTablePrimaryKeyFields(updateExpression, request);
+
+        if (conditionExpression != null) {
+            conditionExpression = convertFieldNameLiteralsToExpressionNames(conditionExpression, request,
+                virtualFieldNames);
+            conditionExpression = makePlaceholdersDistinct(updateExpression, conditionExpression, request);
```

Comment:
To make our lives easier, if the condition expression has overlapping attribute name or value references with the update expression, create more placeholder references so there is no overlap

Reply:
Why is this helpful? I.e., what problem does it create to have overlapping attribute name/value placeholders?

Reply:
Say that in the virtual update request, the update expression is `SET #fieldA = :value1`, and the condition expression is `#fieldB = :value1`, with attribute name map `{fieldA -> A, fieldB -> B}` and attribute value map `{value1 -> x}`. The virtual table has 2 GSIs defined, GSI1 has HK=A, and GSI2 has HK=C and RK=B.

Mapping the update expression, we'd apply the GSI1 FieldMapping for field A, giving us `SET #fieldA = :value1`, with name map `{fieldA -> gsi1HK}`, and value map `{value1 -> ctx/table/x}`.

Mapping the condition expression, if we did this independently without considering the update expression, we'd apply the GSI2 mapping for B, giving us `#fieldB = :value1`, with name map `{#fieldB -> gsi2RK}`, and value map `{value1 -> x}`.

But then we have conflicting value mappings for the `value1` placeholder and we'd have to deal with this somehow. I think we could have a similar scenario where the same field placeholder is used in the two expressions. Making the two expressions not share any field or value placeholders in the first place would make this easy and we can transform the two expressions independently.

Reply:
Thanks that helps a lot!